### PR TITLE
Reveal secrets with a hack #312

### PR DESCRIPTION
**Pull request by louwers**
To be able to store all secrets in a password manager: #312

This temporary workflow will write them to a file.

### CODE REVIEW

**Comment by wipfli** [comment]
Thanks for looking into this, @louwers. I wonder if we need all of the extracted secrets. Some of them might be not used anymore. Can you make a list of the secrets you would like to extract and say what you think they are used for? Thanks!

**Comment by louwers** [comment]
@wipfli Yes I can do that, but it would be helpful to merge this first so I can tell which ones are still usable.

**Comment by wipfli** [comment]
Please make the list first. I think the google cloud tokens should just be deleted.

**Comment by wipfli** [comment]
@acalcutt Do you know how the Node pre gyp token is used?

**Comment by wipfli** [comment]
Maybe the node token is actually a personal access token of @birkskyum?

**Comment by birkskyum** [comment]
I believe NODE_PRE_GYP_GITHUB_TOKEN is linked to my GitHub user. I made the token to allow the node release workflow to push a commit with the new version number.

**Comment by louwers** [comment]
GLCLOUD_SERVICE_JSON is needed for the Android Firebase instrumentation tests.

GPG_KEY_CONTENTS is needed for an Android release. I can't find any usages of GPG_PASSPHRASE.

MAPLIBRE_DEVELOPER_CONFIG_XML is neede to make a Test APK.

NODE_PRE_GYP_GITHUB_TOKEN: see comment above, since it is used I think it makes sense to reveal and use it.


BINTRAY_USER / BINTRAY_API_KEY do not seem to be used. MAPBOX_DEVELOPER_CONFIG_XML is not needed it seems. If you want, remove them before merging this temporary workflow.

**Comment by wipfli** [comment]
OK thanks, can you remove the unneeded ones?

**Comment by louwers** [comment]
@wipfli Done.

File: .github/workflows/reveal-secrets.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  build:
+    runs-on: [self-hosted, macOS, ARM64]
+    defaults:
+      run:
+        working-directory: platform/ios
+        shell: bash
+    steps:
+      - name: Write secrets to file
+        run: 
+          echo "DIST_GITHUB_TOKEN=${{ secrets.DIST_GITHUB_TOKEN}}\nGLCLOUD_SERVICE_JSON=${{ secrets.GLCLOUD_SERVICE_JSON }}\nGPG_KEY_CONTENTS=${{ secrets.GPG_KEY_CONTENTS }}\nMAPLIBRE_DEVELOPER_CONFIG_XML=${{ secrets.MAPLIBRE_DEVELOPER_CONFIG_XML }}\nNODE_PRE_GYP_GITHUB_TOKEN=${{ secrets.NODE_PRE_GYP_GITHUB_TOKEN }}\n" > /Users/ec2-user/secrets
```
Comment (wipfli) [inline .github/workflows/reveal-secrets.yml:16]:
```suggestion
          echo "DIST_GITHUB_TOKEN=${{ secrets.DIST_GITHUB_TOKEN}}\nGLCLOUD_SERVICE_JSON=${{ secrets.GLCLOUD_SERVICE_JSON }}\nGPG_KEY_CONTENTS=${{ secrets.GPG_KEY_CONTENTS }}\nMAPLIBRE_DEVELOPER_CONFIG_XML=${{ secrets.MAPLIBRE_DEVELOPER_CONFIG_XML }}\n" > /Users/ec2-user/secrets
```

Comment (wipfli) [reply .github/workflows/reveal-secrets.yml:16]: Since we do not need to read @birkskyum's PAT...

File: .github/workflows/reveal-secrets.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  build:
+    runs-on: [self-hosted, macOS, ARM64]
+    defaults:
+      run:
+        working-directory: platform/ios
+        shell: bash
+    steps:
+      - name: Write secrets to file
+        run: 
+          echo "DIST_GITHUB_TOKEN=${{ secrets.DIST_GITHUB_TOKEN}}\nGLCLOUD_SERVICE_JSON=${{ secrets.GLCLOUD_SERVICE_JSON }}\nGPG_KEY_CONTENTS=${{ secrets.GPG_KEY_CONTENTS }}\nMAPLIBRE_DEVELOPER_CONFIG_XML=${{ secrets.MAPLIBRE_DEVELOPER_CONFIG_XML }}\n" > /Users/ec2-user/secrets
```
Comment (wipfli) [inline .github/workflows/reveal-secrets.yml:16]:
```suggestion
          echo "GLCLOUD_SERVICE_JSON=${{ secrets.GLCLOUD_SERVICE_JSON }}\nGPG_KEY_CONTENTS=${{ secrets.GPG_KEY_CONTENTS }}\nMAPLIBRE_DEVELOPER_CONFIG_XML=${{ secrets.MAPLIBRE_DEVELOPER_CONFIG_XML }}\n" > /Users/ec2-user/secrets
```

Comment (wipfli) [reply .github/workflows/reveal-secrets.yml:16]: This seems to be another PAT which we should not extract.

**Review by wipfli — APPROVED** [review]
Looks good to me now. Please merge, run once, and then remove this workflow again in a follow-up pull request. Thanks!

**Pull request merged**